sched/Kconfig

```diff
@@ -1826,3 +1826,10 @@ config SCHED_BACKTRACE
 		using the information automatically generated by the
 		compiler or architecture specific approach when ARCH_HAVE_BACKTRACE
 		is selected
+
+config TASK_LIST_OPTIMIZE
+	bool "Enable task list optimize"
+	default n
+	---help---
+		Set to enable task lists optimize.
```

Comment:
This needs a more descriptive explanation. Please answer the question what it does when to use it, why to use it and how it works.